### ISSUE TITLE
Additional inference measures

### ISSUE THREAD

**Issue by pcomw** [Labo-Lacourse/stepmix#46]
Once again, thank you for the wonderful work. 

I hope to switch over to Stepmix from MPlus, but a few tests that my group uses to evaluate models with different numbers of classes in the package, and I was curious about the roadmap. 

In the future, are there plans to add other inference measures to the stepmix class? I am thinking in particular of other IC and LRT-type stats:

1. Sample-size adjusted BIC, e.g., `-2 * model.score(X) * X.shape[0] + model.n_parameters * np.log((X.shape[0] + 2) / 24)`

2. CAIC, e.g., `-2 * model.score(X) * X.shape[0] + model.n_parameters * (np.log(X.shape[0]) + 1)` 

3. Bootstrap likelihood ratio test (BLRT). E.g., page 543 of https://doi.org/10.1080/10705510701575396

4. Possibly also the Lo–Mendell–Rubin (LMR) and/or Vuong–Lo–Mendell–Rubin (VLMR) tests.

The IC stats are simple enough to implement, but the BLRT would take a little time. 

Thanks again

**Comment by sachaMorin** [comment]
Thanks for the suggestions! Sample-size adjusted BIC and CAIC seem easy enough. I'm not 100% about BLRT but I would like to hear StepMix power user @FelixLaliberte's thoughts on this. Is this similar to what you are working on?

**Comment by FelixLaliberte** [comment]
Hi @pcomw,

I think all the inference measures you suggest are commonly used and would be much appreciated by users. Thanks for the suggestions!

@sachaMorin: To answer your question, the BLRT would indeed be very useful.

**Comment by sachaMorin** [comment]
Reviewing [this paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3904803/#R26), it seems the suggested Sample-Adjusted BIC was missing an `X.shape[0]` in the log. The current suggested implementation is

```python3
        n = X.shape[0]

        return -2 * self.score(X, Y) * n + self.n_parameters * np.log(
            n * ((n + 2) / 24)
        )
```

**Comment by pcomw** [comment]
I agree, it seems in the paper you linked that it should be `n * ((n + 2) / 24)`

However, in the MPlus output, there is a line of text that reads, `(n* = (n + 2) / 24)`, which implies that the asterisk isn't a sign of multiplication, but a signifier of a different value. Similarly, on page 545 of https://doi.org/10.1080/10705510701575396, the text defines SABIC as replacing _n_ in the original BIC calculation with _n*_, as seen here:

<img width="503" alt="Annotation 2023-08-01 090030" src="https://github.com/Labo-Lacourse/stepmix/assets/126023137/62b115da-41a9-4b9c-916a-16846ae0864d">

I suppose that the 1987 Sclove paper is the proper resource to check, but I don't have access to it.

**Comment by sachaMorin** [comment]
CAIC and SABIC are available as of version [2.0.0](https://github.com/Labo-Lacourse/stepmix/releases/tag/2.0.0).

**Comment by sachaMorin** [comment]
BLRT is available as of version [2.2.0](https://github.com/Labo-Lacourse/stepmix/releases/tag/2.2.0).  See the [updated tutorial](https://colab.research.google.com/drive/1btXHCx90eCsnUlQv_yN-9AzKDhJP_JkG?usp=drive_link).

**Comment by Acgholic** [comment]
We would very appreciate if the LMR is available in the future.
